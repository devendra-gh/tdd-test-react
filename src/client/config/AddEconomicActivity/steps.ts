export interface IStep {
  name: string;
  subSteps?: string[];
}

export const addEconomicActivitySteps: IStep[] = [
  { name: 'addEconomicActivity.addActivity' },
  { name: 'addEconomicActivity.getDEDApproval' },
];
