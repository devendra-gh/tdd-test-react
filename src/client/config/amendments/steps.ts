export interface IStep {
  name: string;
  subSteps?: string[];
  id?: string;
}

export const steps: IStep[] = [
  {
    name: 'process.selectLicenceNumber',
  },
  {
    name: 'process.selectAmendmentType',
  },
  {
    id: 'amendments',
    name: 'process.makeAmendment',
  },
  {
    name: 'process.getDEDApproval',
  },
  {
    name: 'process.makePayment',
  },
  {
    name: 'process.downloadLicence',
  },
];

export default steps;
