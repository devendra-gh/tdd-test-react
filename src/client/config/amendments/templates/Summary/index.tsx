import React from 'react';
import PropTypes from 'prop-types';
import { withTemplateHooks, IVariables } from '@tamm/app-composer';
import routePropTypes from '@tamm/app-composer/client/propTypes/route';
import Table from '@tamm/ui-lib-v2-table';
import Container from 'client/containers';
import Sidebar from 'client/templates/AmendmentsSidebar';
import Total from '@tamm/ui-lib-v2-total';
import Button from '@tamm/ui-lib-v2-button';
import Notice from '@tamm/ui-lib-v2-notice';
import { InformationCircleFilled } from '@tamm/ui-lib-v2-icon/Icons';
import Informational from '@tamm/ui-lib-v2-informational-template';
import Alert from '@tamm/ui-lib-v2-alert';

/**
 * Summary template
 * @param {Object} props
 * @returns {JSX}
 */

const SummaryTemplate = (props: IVariables) => {
  const { list, amendmentServerError } = props;

  const icon = () => (
    <InformationCircleFilled className="ui-lib-notice__icon" />
  );
  const { i18n, pay } = props;

  return (
    <Container
      locale={props.locale}
      sidebar={
        <Sidebar
          currentStep={props.currentStep}
          currentSubStep={props.currentSubStep}
          i18n={props.i18n}
          steps={props.steps}
          stepsStatus={props.stepsStatus}
          showSidebar={props.showSidebar}
        />
      }
    >
      <Notice
        buttons={props.buttons.map((button: any) => ({
          'aria-label': `button-${button.variant}`,
          label: i18n(button.label),
          uiType: button.uiType ? button.uiType : 'primary',
          withArrow: props.withArrow,
          // onClick: () => {
          //   if (button.link) {
          //     props.history.push(button.link);
          //   } else if (button.onClick) {
          //     button.onClick(props);
          //   }
          // },
          size: 'medium',
        }))}
        content={
          <Informational>
            {props.subTitle && (
              <h5 className="titleColor">{i18n(props.subTitle)}</h5>
            )}
            {props.content && (
              <p className="subtitleColor">{i18n(props.content)}</p>
            )}
          </Informational>
        }
        status="success"
        icon={icon}
        tags={
          props.tags &&
          props.tags.map((tag: any) => ({
            ...tag,
            label: i18n(tag.label),
          }))
        }
        title={i18n(props.title)}
      />
      <div className="height-20" />
      <hr />
      <div className="height-20" />
      {amendmentServerError && (
        <div id="errorMessage">
          <Alert message={amendmentServerError} status="error" />
          <div className="height-20" />
        </div>
      )}
      <Informational>
        <h3 className="titleColor">{i18n('payment.summary')}</h3>
        <p className="subtitleColor">{i18n('payment.description')}</p>
      </Informational>
      <div className="height-20" />
      <div className="summary">
        <div className="summary__group">
          <Table
            i18n={i18n}
            clickable={false}
            columns={list.columns}
            disabledSelectionVisible={false}
            headerHidden={false}
            items={list.items}
            // onClick={function noRefCheck(){}}
            // onSelectionChange={function noRefCheck(){}}
            // onToggle={function noRefCheck(){}}
            selectable={false}
            size="default"
            title={props.i18n('global.fees')}
          />
        </div>
        <div className="height-20" />
        <div className="summary__group">
          <Total unit={props.i18n('global.aed')} value={props.totalSection} />
        </div>
        <div className="height-20" />
        <div className="summary__group">
          <Button
            aria-label="pay"
            label={i18n('button.pay')}
            onClick={pay(props)}
            // uiType={btn.uiType}
            withArrow
          />
        </div>
      </div>
      <div className="height-100" />
    </Container>
  );
};

SummaryTemplate.prototypes = {
  ...routePropTypes,
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  description: PropTypes.string,
  list: PropTypes.instanceOf(Object).isRequired,
  button: PropTypes.instanceOf(Object).isRequired,
};

export default withTemplateHooks(SummaryTemplate);
