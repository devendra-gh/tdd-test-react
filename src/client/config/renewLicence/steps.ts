// list of substeps ids
export interface IStep {
  name: string;
  subSteps?: string[];
}
export const FORM_STEP_1 = 'step1.title';
export const FORM_STEP_2 = 'step2.title';
export const FORM_STEP_3 = 'step3.title';
export const FORM_STEP_4 = 'step4.title';
export const FORM_STEP_5 = 'step5.title';
export const FORM_STEP_6 = 'step6.title';
export const minimumSteps: IStep[] = [
  { name: FORM_STEP_1 },
  { name: FORM_STEP_2 },
];
const steps: IStep[] = [
  { name: FORM_STEP_1 },
  { name: FORM_STEP_2 },
  { name: FORM_STEP_3 },
  { name: FORM_STEP_4 },
  { name: FORM_STEP_5 },
  { name: FORM_STEP_6 },
];

export default steps;
