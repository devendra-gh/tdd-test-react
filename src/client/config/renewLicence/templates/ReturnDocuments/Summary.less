.summary {
  margin-bottom: 4rem;

  &__group {
    margin-top: 4rem;
  }
}
