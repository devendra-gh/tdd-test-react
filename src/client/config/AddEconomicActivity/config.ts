import '@tamm/ui-lib-v2-styles/common.less';
import '@tamm/ui-lib-v2-styles/colors.less';
import './override.less';
import { IVariables } from '@tamm/app-composer';
import { addEconomicActivitySteps } from './steps';
import pages from './pages';
import { getVariables } from './services/bpm';

export const initialState: IVariables = {
  fileUploadData: {
    documents: {
      activitySupportingDoc: null,
    },
  },

  instanceId: '',
  businessKey: '',

  formData: {
    arabicActivityName: '',
    englishActivityName: '',
    arabicActivityDescription: '',
    englishActivityDescription: '',
    name: '',
    email: '',
    mobileNumber: '',
  },
  helperData: {
    isSubmitted: false,
  },
  newActivityApiData: {
    submittedOn: '',
    altId: '',
  },
  statusRecieved: false,
  stepsStatus: {},
  addEconomicActivitySteps,
};

const redirectUrl =
  window.location.href.indexOf('stage.tamm.abudhabi') !== -1
    ? 'https://stage.tamm.abudhabi/'
    : 'https://www.tamm.abudhabi/';

const config = {
  appName: 'newActivityService',
  // configuration/app version, please keep updateding
  // for now redux counts on this to disregard the old state
  version: '0.6',
  // default variables, for now title is used
  defaults: {
    title: 'default.title',
  },
  getVariables,
  // initial redux state, you can configure per your project need,
  initialState,
  header: {
    template: 'header',
    props: {
      aspectsOfLifeType: 'manage_business',
      breadcrumbs: [
        {
          label: 'breadcrumb.home',
          link: `${redirectUrl}`,
          linkTarget: '_self',
        },
        {
          label: 'breadcrumb.digitalServices',
          link: `${redirectUrl}tamm-centers-services`,
          linkTarget: '_self',
        },
        {
          label: 'breadcrumb.dedServices',
          link: `${redirectUrl}tamm-centers-services/department-of-economic-development`,
          linkTarget: '_self',
        },
      ],
    },
    state: {
      mapState: ['user', 'locale', 'title'],
      mapDispatch: ['user', 'locale'],
    },
  },
  footer: {
    template: 'footer',
    state: {
      mapState: ['user'],
    },
  },
  // pages of the application
  pages,
};

export default config;
