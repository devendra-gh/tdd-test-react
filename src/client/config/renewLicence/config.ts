import bpm from 'client/services/bpm';
// import { getSmartpassData } from 'client/utils/appData';
import { IVariables } from '@tamm/app-composer';
import steps, {
  minimumSteps,
  FORM_STEP_2,
} from 'client/config/renewLicence/steps';
import pages from './pages';

// const smartpassData = getSmartpassData();

export const initialState: IVariables = {
  nocForm: {},
  fileUploadData: {
    documents: {
      thawtheeq: null,
      noc: null,
    },
  },
  form: {
    licenceNo: '',
    licenceExpiryDate: '',
    daysPendingForLicenceExpiry: null,
  },
  stepsStatus: {},
  steps,
  minimumSteps,
  instanceId: '',
  businessKey: '',
  reachedPayment: false, // 2 waiting pages toggled using this
  isTawtheeqRequired: true,
  licenceSubmitPage: FORM_STEP_2,
  paymentAmount: '0',
};

const redirectUrl =
  window.location.href.indexOf('stage.tamm.abudhabi') !== -1
    ? 'https://stage.tamm.abudhabi/'
    : 'https://www.tamm.abudhabi/';

const config = {
  appName: 'renewLicence',
  // configuration/app version, please keep updateding
  // for now redux counts on this to disregard the old state
  version: '1',
  // default variables, for now title is used
  defaults: {
    title: 'Renew Licence',
  },
  getVariables: bpm.getVariables,
  // initial redux state, you can configure per your project need,
  initialState,
  header: {
    template: 'header',
    props: {
      aspectsOfLifeType: 'business-management',
      breadcrumbs: [
        {
          label: 'breadcrumb.home',
          link: `${redirectUrl}`,
          linkTarget: '_self',
        },
        {
          label: 'breadcrumb.digitalServices',
          link: `${redirectUrl}tamm-centers-services`,
          linkTarget: '_self',
        },
        {
          label: 'breadcrumb.dedServices',
          link: `${redirectUrl}tamm-centers-services/department-of-economic-development`,
          linkTarget: '_self',
        },
      ],
    },
    state: {
      mapState: ['user', 'locale', 'title', 'form', 'minimumSteps'],
      mapDispatch: ['user', 'locale'],
    },
    // hideForPaths: ['/login'],
  },
  footer: {
    template: 'footer',
    state: {
      mapState: ['user'],
    },
    // hideForPaths: ['/login'],
  },
  // pages of the application
  pages,
};

export default config;
