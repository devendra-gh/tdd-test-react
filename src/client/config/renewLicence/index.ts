import config from './config';
import fetchState from './state';
import templates from './templates';
import translations from './localisation';
import { PATH_LICENCE_SUMMARY } from './routes';

const skipFetchState: string[] = ['/', PATH_LICENCE_SUMMARY];

export { config, fetchState, templates, skipFetchState, translations };
