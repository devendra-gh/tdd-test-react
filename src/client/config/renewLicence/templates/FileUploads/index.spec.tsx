import fetch from 'client/services/fetch';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router';
import { IVariables } from '@tamm/app-composer';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import { steps } from 'client/config/steps';
import Summary from './index';

jest.mock('client/services/fetch');

export interface IFileUploadProps {
  label: string;
  onChange: Function;
  onRemove: Function;
}

jest.mock(
  '@tamm/ui-lib-v2-file-upload',
  () => ({ onChange, label, onRemove }: IFileUploadProps) => {
    const onChangeFile = (e: IVariables) => {
      if (e.target.files.length) onChange(e.target.files);
      else onRemove(e.target.files);
    };

    return (
      <>
        <label htmlFor="test-upload">{label}</label>
        <input
          data-testid="file-upload-input"
          name="test-upload"
          id="test-upload"
          type="file"
          onChange={onChangeFile}
        />
      </>
    );
  },
);

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('config/templates/Summary', () => {
  let props: IVariables;
  let mockFetch: any;
  beforeEach(() => {
    mockFetch = fetch;
    props = {
      fileUploadData: {
        documents: {
          thawtheeq: [{}],
          noc: [{ loading: true }],
        },
      },
      inputGroups: [
        {
          name: 'thawtheeq',
          twoColumns: true,
          stateKey: 'documents',
          fields: [
            {
              'aria-label': 'file upload',
              elementType: 'fileUpload',
              name: 'thawtheeq',
              accept: ['application/pdf'],
              label: 'field.supportingDocuments',
              maxSize: 5e6,
              help: 'validationMessage.required',
            },
            {
              'aria-label': 'file upload',
              elementType: 'fileUpload',
              name: 'noc',
              accept: ['application/pdf'],
              label: 'field.supportingDocuments',
              maxSize: 5e6,
              help: 'validationMessage.required',
            },
          ],
        },
      ],
      submitLicence: {
        data: {
          thawtheeq: [{}],
        },
      },
      i18n: jest.fn(),
      title: 'summary',
      subTitle: 'subTitle',
      description: 'desc',
      locale: 'en',
      currentStep: 'economic_licence',
      currentSubStep: 'payment',
      buttons: [
        {
          label: 'btn',
          onClick: jest.fn(),
        },
      ],
      steps,
      stepStatus: {
        economic_name: 'economic_name',
        'economic_name.payment': 'economic_name.payment',
      },
      list: {
        labelHeading: 'label',
        valueHeading: 'value',
        listDetails: {
          label: 'label',
          value: 'value',
        },
      },
      totalSection: 1000,
      isTawtheeqRequired: true,
      isNOCRequired: true,
      handleChange: jest.fn(),
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render with all props', () => {
    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <Summary {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with all props, when file loading', () => {
    props.fileUploadData = {
      documents: {
        thawtheeq: [{ loading: true }],
      },
    };

    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <Summary {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with all props, when file undefined', () => {
    props.fileUploadData = {
      documents: {},
    };

    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <Summary {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with all props, when startShowingErrors true', () => {
    props.startShowingErrors = true;
    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <Summary {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onFileUpload when onChange triggered', () => {
    mockFetch.mockImplementation(() => {
      return Promise.resolve({
        data: {
          uploadedFileDetails: { s3FilePath: 's3', nameOfFile: 'name' },
        },
      });
    });
    const { getAllByTestId } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <Summary {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    const fileUpload = getAllByTestId('file-upload-input')[0];
    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
      type: 'image/png',
    });
    fireEvent.change(fileUpload);
    // fireEvent.click(, undefined);
    expect(props.handleChange).toHaveBeenCalled();

    Object.defineProperty(fileUpload, 'files', {
      value: [file],
    });
    fireEvent.change(fileUpload);
    expect(props.handleChange).toHaveBeenCalled();
  });

  it('should call onFileUpload when onChange triggered, and api fails ', () => {
    mockFetch.mockImplementation(() => {
      return Promise.reject(new Error());
    });

    const { getAllByTestId } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <Summary {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    const fileUpload = getAllByTestId('file-upload-input')[0];
    const file = new File(['(⌐□_□)'], 'chucknorris.png', {
      type: 'image/png',
    });
    fireEvent.change(fileUpload);
    expect(props.handleChange).toHaveBeenCalled();

    Object.defineProperty(fileUpload, 'files', {
      value: [file],
    });
    fireEvent.change(fileUpload);
    expect(props.handleChange).toHaveBeenCalled();
  });
});
