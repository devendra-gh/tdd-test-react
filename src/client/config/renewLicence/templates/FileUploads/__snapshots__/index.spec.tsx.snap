// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/templates/Summary should render with all props 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <form
    class="ui-lib-form qa-form"
    id=""
    name="file-upload-section"
  >
    
    <div>
      <fieldset
        class="ui-lib-form__fieldset"
      >
        
        <div
          class="ui-lib-form__fieldset-content ui-lib-form__fieldset-content_default-gap"
        >
          <label
            for="test-upload"
          />
          <input
            data-testid="file-upload-input"
            id="test-upload"
            name="test-upload"
            type="file"
          />
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;

exports[`config/templates/Summary should render with all props, when file loading 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <form
    class="ui-lib-form qa-form"
    id=""
    name="file-upload-section"
  >
    
    <div>
      <fieldset
        class="ui-lib-form__fieldset"
      >
        
        <div
          class="ui-lib-form__fieldset-content ui-lib-form__fieldset-content_default-gap"
        >
          <label
            for="test-upload"
          />
          <input
            data-testid="file-upload-input"
            id="test-upload"
            name="test-upload"
            type="file"
          />
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;

exports[`config/templates/Summary should render with all props, when file undefined 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <form
    class="ui-lib-form qa-form"
    id=""
    name="file-upload-section"
  >
    
    <div>
      <fieldset
        class="ui-lib-form__fieldset"
      >
        
        <div
          class="ui-lib-form__fieldset-content ui-lib-form__fieldset-content_default-gap"
        >
          <label
            for="test-upload"
          />
          <input
            data-testid="file-upload-input"
            id="test-upload"
            name="test-upload"
            type="file"
          />
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;

exports[`config/templates/Summary should render with all props, when startShowingErrors true 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <form
    class="ui-lib-form qa-form"
    id=""
    name="file-upload-section"
  >
    
    <div>
      <fieldset
        class="ui-lib-form__fieldset"
      >
        
        <div
          class="ui-lib-form__fieldset-content ui-lib-form__fieldset-content_default-gap"
        >
          <label
            for="test-upload"
          />
          <input
            data-testid="file-upload-input"
            id="test-upload"
            name="test-upload"
            type="file"
          />
        </div>
      </fieldset>
    </div>
  </form>
</div>
`;
